src/index.js

```diff
@@ -86,6 +86,12 @@ function fetch (uri, opts) {
           reject(new FetchError(`redirect location header missing at: ${request.url}`, 'invalid-redirect'))
           return
         }
+        // Remove authorization if changing hostnames (but not if just
+        // changing ports or protocols).  This matches the behavior of request:
+        // https://github.com/request/request/blob/b12a6245/lib/redirect.js#L134-L138
+        if (url.parse(request.url).hostname !== url.parse(res.headers.location).hostname) {
```

Comment:
Correction!

The [`Location` header](https://tools.ietf.org/html/rfc7231#section-7.1.2) accepts a [`URI-Reference`](https://tools.ietf.org/html/rfc3986#section-4.1). That is, _either_ a relative ref, or a fully-qualified URI (with ports). `www.foo.com` is invalid. I kinda wonder if anyone sends it that way, though.